apps/src/lib/node/ledger/shell/vote_extensions.rs

```diff
@@ -344,6 +371,104 @@ where
+                Ok(tx) => tx,
+                Err(err) => {
+                    tracing::warn!(
+                        ?err,
+                        "Failed to deserialize tx in \
+                         deserialize_vote_extensions"
+                    );
+                    return None;
+                }
+            };
+            match process_tx(tx).ok()? {
+                TxType::Protocol(ProtocolTx {
+                    tx:
+                        ProtocolTxType::EthEventsVext(_)
+                        | ProtocolTxType::BridgePoolVext(_),
+                    ..
+                }) => {
+                    // mark tx for inclusion or it is skipped
+                    // if the bridge is inactive
+                    protocol_tx_indices.insert(index);
+                    self.storage.is_bridge_active().then_some(tx_bytes.clone())
+                }
+                TxType::Protocol(ProtocolTx {
+                    tx: ProtocolTxType::ValSetUpdateVext(ext),
+                    ..
+                }) => {
+                    // mark tx, so it's skipped when
+                    // building the batch of remaining txs
+                    protocol_tx_indices.insert(index);
+
+                    // only include non-stale validator set updates
+                    // in block proposals. it might be sitting long
+                    // enough in the mempool for it to no longer be
+                    // relevant to propose (e.g. the new epoch was
+                    // installed before this validator set update got
+                    // a chance to be decided). unfortunately, we won't
+                    // be able to remove it from the mempool this way,
+                    // but it will eventually be evicted, getting replaced
+                    // by newer txs.
+                    (ext.data.signing_epoch == current_epoch)
+                        .then(|| tx_bytes.clone())
+                }
+                _ => None,
+            }
+        })
+    }
+
+    /// Deserializes `vote_extensions` as [`VoteExtension`] instances, filtering
+    /// out invalid data, and splits these into [`ethereum_events::Vext`]
+    /// and [`validator_set_update::Vext`] instances.
+    #[cfg(feature = "abcipp")]
+    #[allow(clippy::type_complexity)]
+    pub fn split_vote_extensions(
+        &self,
+        vote_extensions: Vec<ExtendedVoteInfo>,
+    ) -> (
+        Option<Vec<Signed<ethereum_events::Vext>>>,
+        Option<Vec<Signed<bridge_pool_roots::Vext>>>,
+        Vec<validator_set_update::SignedVext>,
+    ) {
+        let mut eth_evs = vec![];
+        let mut bp_roots = vec![];
+        let mut valset_upds = vec![];
+        let bridge_active = self.storage.is_bridge_active();
+
+        for ext in deserialize_vote_extensions(vote_extensions) {
+            if let Some(validator_set_update) = ext.validator_set_update {
+                valset_upds.push(validator_set_update);
+            }
```

Comment:
validator set updates should also be excluded from proposals, but they must be included an epoch before the bridge is started. therefore, `is_bridge_active()` needs an `Epoch` param. we should check if the bridge is active at the next epoch, to include valset upds

Reply:
ah nevermind, just realized we have `check_bridge_status()`. this should be what we should be querying, then

Reply:
That code isn't actually changed. It just registered as changed to github somehow.